packages/e2e-tests/test/e2e-streams.spec.ts

```diff
@@ -0,0 +1,328 @@
+      );
+      return this.skip();
+    }
+  });
+
+  describe('basic stream processor operations', function () {
+    let processorName = '';
+    let db: Db;
+    let collection: Collection<Document>;
+    let client: MongoClient;
+
+    beforeEach(async function () {
+      shell = this.startTestShell({
+        args: [
+          STREAMS_E2E_SPI_CONNECTION_STRING,
+          '--tls',
+          '--authenticationDatabase admin',
+          '--username',
+          STREAMS_E2E_DB_USER,
+          '--password',
+          STREAMS_E2E_DB_PASSWORD,
+        ],
+        removeSigintListeners: true,
+      });
+      await shell.waitForPromptOrExit({ timeout: 60_000 });
+
+      processorName = `spi${new bson.ObjectId().toHexString()}`;
+      client = await MongoClient.connect(
+        STREAMS_E2E_CLUSTER_CONNECTION_STRING,
+        {}
+      );
+      db = client.db(processorName);
+      const collectionName = 'processedData';
+      collection = db.collection(collectionName);
+
+      // this stream processor reads from the sample stream and inserts documents into an Atlas database
+      const sourceStage = {
+        $source: {
+          connectionName: 'sample_stream_solar',
+        },
+      };
+
+      const mergeStage = {
+        $merge: {
+          into: {
+            connectionName: 'testClusterConnection',
+            db: processorName,
+            coll: collectionName,
+          },
+        },
+      };
+
+      const aggPipeline = [sourceStage, mergeStage];
+
+      const createResult = await shell.executeLine(
+        `sp.createStreamProcessor("${processorName}", ${JSON.stringify(
+          aggPipeline
+        )})`
+      );
+      expect(createResult).to.include(
+        `Atlas Stream Processor: ${processorName}`
+      );
+    });
+
+    afterEach(async function () {
+      try {
+        await db.dropDatabase();
+        await client.close();
+
+        const result = await shell.executeLine(`sp.${processorName}.drop()`);
+        expect(result).to.include(`{ ok: 1 }`);
+      } catch (err: any) {
+        console.error(
+          `Could not clean up stream processor ${processorName}:`,
+          err
+        );
+      }
+    });
+
+    it('can list stream processors', async function () {
+      const listResult = await shell.executeLine(`sp.listStreamProcessors()`);
+      // make sure the processor created in the beforeEach is present
+      expect(listResult).to.include(`name: '${processorName}'`);
+    });
+
+    it('can start and stop a stream processor', async function () {
+      // this should be a unique collection for this test run, so no data to start
+      const initialDocsCount = await collection.countDocuments();
+      expect(initialDocsCount).to.eq(0);
+
+      const startResult = await shell.executeLine(
+        `sp.${processorName}.start()`
+      );
+      expect(startResult).to.include('{ ok: 1 }');
+
+      // sleep for a bit to let the processor do stuff
+      await sleep(500);
```

Comment:
Would it make sense to instead wait using `eventually()` for the assertion below on the first `countDocuments()`'s return value to be true? That should help with avoiding flakiness from this bit